homeassistant/components/input_datetime.py

```diff
@@ -0,0 +1,237 @@
+"""
+Component to offer a way to select a date and / or a time.
+
+For more details about this component, please refer to the documentation
+at https://home-assistant.io/components/input_datetime/
+"""
+import asyncio
+import logging
+import datetime
+
+import voluptuous as vol
+from voluptuous.humanize import humanize_error
+
+from homeassistant.const import ATTR_ENTITY_ID, CONF_ICON, CONF_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.restore_state import async_get_last_state
+from homeassistant.util import dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'input_datetime'
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+
+CONF_HAS_DATE = 'has_date'
+CONF_HAS_TIME = 'has_time'
+CONF_INITIAL = 'initial'
+
+ATTR_DATE = 'date'
+ATTR_TIME = 'time'
+
+SERVICE_SET_DATETIME = 'set_datetime'
+
+SERVICE_SET_DATETIME_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+    vol.Optional(ATTR_DATE): cv.date,
+    vol.Optional(ATTR_TIME): cv.time,
+})
+
+SERVICE_WITH_DATE_SUBSCHEMA = {
+    vol.Required(ATTR_DATE): cv.date
+}
+
+SERVICE_WITH_TIME_SUBSCHEMA = {
+    vol.Required(ATTR_TIME): cv.time
+}
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        cv.slug: vol.All({
+            vol.Optional(CONF_NAME): cv.string,
+            vol.Required(CONF_HAS_DATE): cv.boolean,
+            vol.Required(CONF_HAS_TIME): cv.boolean,
+            vol.Optional(CONF_ICON): cv.icon,
+            vol.Optional(CONF_INITIAL,
+                         default=datetime.datetime(1970, 1, 1)): cv.datetime,
```

Comment:
That should be not have a default value. If we have a initial value we reset every boot this to initial state like other input types.